src/System.IdentityModel.Tokens.Jwt/JwtPayload.cs

```diff
@@ -350,11 +350,21 @@ public string CHash
         /// Gets the 'value' of the 'expiration' claim { exp, 'value' }.
         /// </summary>
         /// <remarks>If the 'expiration' claim is not found OR could not be converted to <see cref="Int32"/>, null is returned.</remarks>
+        [Obsolete("`int? JwtPayload.Exp` is deprecated and will be removed in a future release. Use `long? JwtPayload.Expiration` instead. For more information, see https://aka.ms/IdentityModel/7-breaking-changes")]
         public int? Exp
         {
             get => GetIntClaim(JwtRegisteredClaimNames.Exp);
         }
 
+        /// <summary>
+        /// Gets the 'value' of the 'expiration' claim { exp, 'value' }.
+        /// </summary>
+        /// <remarks>If the 'expiration' claim is not found OR could not be converted to <see cref="long"/>, null is returned.</remarks>
+        public long? Expiration
```

Comment:
Don't we want to use a DateTime (or a DateTimeOffset)?

Reply:
@jmprieur we have ValidFrom and ValidTo as DateTime.
This property was for users that wanted the epoch time.

Reply:
Given it's in Wilon7 (which has breaking changes), can't we just return `long` then? instead of `int` ?

Reply:
@jmprieur we will be returning `long?` instead of `int?`, if I understand your question correctly